src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```diff
@@ -171,14 +171,12 @@ class UnionMangas(private val langOption: LanguageOption) : HttpSource() {
     }
 
     override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
-        if (query.startsWith(slugPrefix)) {
-            val mangaUrl = query.substringAfter(slugPrefix)
-            return client.newCall(GET("$baseUrl/${langOption.infix}/$mangaUrl", headers))
+        if (query.startsWith(SEARCH_PREFIX)) {
+            val url = "$baseUrl/${langOption.infix}/${query.substringAfter(SEARCH_PREFIX)}"
+            return client.newCall(GET(url, headers))
                 .asObservableSuccess().map { response ->
-                    val manga = mangaDetailsParse(response).apply {
-                        url = mangaUrl
-                    }
-                    MangasPage(listOf(manga), false)
+                    val mangas = try { listOf(mangaDetailsParse(response)) } catch (_: Exception) { emptyList() }
```

Comment:
no need to catch exception

Reply:
> no need to catch exception

When the  deep link is used, all languanges are requested, but the Source returns sucess (200, Even if it can't find anything) and throws an exception in the parse content